### PR TITLE
docs: [cmd] add design doc for cli configuration

### PR DESCRIPTION
**Pull request by dymart**
time to review and update design docs for cli configuration

### CODE REVIEW

File: docs/design/001-cli-config-structure.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+### Feature
+
+A feature refers to some functionality that is exposed through a plugin. There can be multiple features in a single plugin.
+
+Features should be exposed independently inside of a plugin so that they can be turned on and off independently.
+
+## Config File Content Structure
+
+The aspect cli config should be broken down into different namespaces that map to each plugin. This structure can be nested so that features of plugins can also use their own namespace structure to control their own features and settings. The core aspect cli will only deal with the top level namespaces and turing those plugins on appropriately. Dealing with the rest is up to the creator of the plugin. It is recommended that the same namespace structure is used to control features of a plugin as well so that it is easy and consistent for users to understand a config file with multiple plugins and update them as needed.
```
Comment (alexeagle) [inline docs/design/001-cli-config-structure.md:88]: This feels like a really early draft. What are these namespaces? I think it's hard to discuss the structure without a little example here.

File: docs/design/001-cli-config-structure.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+This should include a relevant namespace nested example with multiple plugins and features:
+
+It is also relevant to know that all features available through the config file should also be flags that can be passed to the aspect cli. One should also consider that features and settings in the config file should be modifiable through a command in the aspect cli that writes to the config file. This is just to say take care in naming and structuring your namespace so that commands are readable on their own.
+
+How to deal with plugin ordering and dependencies?
+This is the lest discussed section. Open for comment. Should we just add a common top level way for aspect cli core to understand which plugins are required for others to run? How does this work with a plugin that has multiple features? Can the feature say it needs the plugin or does the top level have to say it requires that plugin? What does the error message and error handling look like here? How do we make sure there is an obvious and actionable error message for a user?
```
Comment (alexeagle) [inline docs/design/001-cli-config-structure.md:95]: Are you looking for help with this part of the design then? Please reach out and be noisy and don't wait for someone to come ask questions in the doc.

File: docs/design/001-cli-config-structure.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+Status: **DRAFT**<br>
+Date: _April 20, 2022_<br>
+Authors: _Dylan Martin \<dylan@aspect.dev\>_<br>
+Reviewers: _..._<br>
+Summary: _How to structure the config files and the their layout for the aspect cli._<br>
+
+---
+
+**This document follows the conventions from [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).**
```
[the remark this answers is not in the document]
Comment (dymart) [reply docs/design/001-cli-config-structure.md:11]: Okay I can remove this, I guess we should also remove it from the template as well then https://github.com/aspect-build/document-templates/blob/main/design_doc_template.md

**Comment by dymart** [comment]
moved to hack md

**Pull request closed without merging**